lib/FreeRTOS/freertos-plus-mqtt/agent_command_pool.c

```diff
@@ -138,27 +152,16 @@ bool Agent_ReleaseCommand( Command_t * pCommandToRelease )
     bool structReturned = false;
 
     /* See if the structure being returned is actually from the pool. */
-    for( i = 0; i < MQTT_COMMAND_CONTEXTS_POOL_SIZE; i++ )
+    if( ( pCommandToRelease >= commandStructurePool ) &&
+        ( pCommandToRelease < ( commandStructurePool + MQTT_COMMAND_CONTEXTS_POOL_SIZE ) ) )
     {
-        if( pCommandToRelease == &( commandStructurePool[ i ] ) )
-        {
-            taskENTER_CRITICAL();
-
-            /* Yes its from the pool.  Clearing it to zero not only removes the old
-             * data it also sets the structure's commandType parameter to NONE to
-             * mark the structure as free again. */
-            memset( ( void * ) pCommandToRelease, 0x00, sizeof( Command_t ) );
-            taskEXIT_CRITICAL();
-
-            /* Give back the counting semaphore after returning the structure so the
-             * semaphore count equals the number of available structures. */
-            xSemaphoreGive( freeCommandStructMutex );
-            structReturned = true;
-
-            LogDebug( ( "Returned Command Context %d to pool", ( int ) i ) );
+        structReturned = Agent_MessageSend( &commandStructQueue, &pCommandToRelease, 0U );
```

Comment:
I didn't handle that since we only add pool structures to the queue, and only after we've already removed them from it. Since the queue was created to hold exactly the number of structures in the pool, we won't be calling this if the queue is full